### PR TITLE
EB-1265 Add new increment method.

### PR DESCRIPTION
**Pull request by eb-palmeida**

### CODE REVIEW

File: src/EBT/CacheClient/Service/Provider/BaseProviderService.php [hunk 1]
```diff
@@ -113,6 +113,33 @@ public function delete($key, array $options = array())
...
+    public function increment($key, $increment = 1, $initialValue = 0, $expiration = null, array $options = array())
+    {
+        /* Validate parameters. */
+        switch (false) {
+            case $this->getValidator()
+                ->isRequiredStringNotEmpty($key):
+            case $this->getValidator()
+                ->isRequiredPositiveInteger($increment):
+            case $this->getValidator()
+                ->isRequiredPositiveInteger($initialValue):
```
Comment (eb-dteixeira) [inline src/EBT/CacheClient/Service/Provider/BaseProviderService.php]: Change to `isRequiredZeroOrPositiveInteger()`.

File: src/EBT/CacheClient/Service/Provider/MemcachedProviderService.php [hunk 2]
```diff
@@ -94,6 +95,50 @@ protected function doDelete($key, array $options = array())
...
+        if ($this->isSuccess()) {
+
+            return new CacheResponse($increment + $initialValue, true, true);
+        }
+
+        /* The key was not stored because is already existed, try to increment a last time. */
+        if ($this->isNotStored()) {
+            $result = $this->client->increment($key, $increment);
+
+            return new CacheResponse($result, true, true);
```
Comment (eb-dteixeira) [inline src/EBT/CacheClient/Service/Provider/MemcachedProviderService.php]: This should be `return new CacheResponse($result, $this->isSuccess(), $this->isSuccess() || $this->isNotFound())`.

File: src/EBT/CacheClient/Service/Provider/MemcachedProviderService.php [hunk 3]
```diff
@@ -94,6 +95,50 @@ protected function doDelete($key, array $options = array())
...
+        }
+
+        /* If the binary protocol is disable we must implement the initial value logic. */
+        $result = $this->client->increment($key, $increment);
+
+        /* In case or success or any error aside "not found" there's nothing more to do. */
+        if ($this->isSuccess() || ! $this->isNotFound()) {
+
+            return new CacheResponse($result, true, true);
```
Comment (eb-dteixeira) [inline src/EBT/CacheClient/Service/Provider/MemcachedProviderService.php:114]: This should be `return new CacheResponse($result, $this->isSuccess(), $this->isSuccess() || $this->isNotFound())`.

File: tools/install-codesniffer.sh [hunk 4]
```diff
@@ -8,4 +8,4 @@ chmod +x ./.git/hooks/pre-commit
 pushd ./vendor/squizlabs/php_codesniffer/CodeSniffer/Standards
```
Comment (eb-dteixeira) [inline tools/install-codesniffer.sh:8]: We could add our commit hooks here since we're altering it.

Comment (eb-palmeida) [inline tools/install-codesniffer.sh:8]: I did so. But then I remembered this is a public repo.

**Pull request merged**